02_configure_host.sh

```diff
@@ -273,36 +270,50 @@ elif [[ "$reg_state" != "running" ]]; then
 fi
 sleep 5
 
+detect_mismatch(){
+  local LOCAL_IMAGE="$1"
+  local REPO_PATH="$2"
+  if [[ -z "${LOCAL_IMAGE}" ]] || [[ "${LOCAL_IMAGE}" == "${REPO_PATH}" ]]; then
+    echo "Local image: ${LOCAL_IMAGE} and repo path: ${REPO_PATH} are matching!"
+  else
+    echo "There is a mismatch between BMO_LOCAL_IMAGE:${LOCAL_IMAGE} and BMO_IMAGE_PATH:${REPO_PATH}"
+    echo "The mismatch could cause hard to debug errors, PLEASE FIX!"
+    exit 1
+  fi
 
+}
 # Clone all needed repositories (CAPI, CAPM3, BMO, IPAM)
+# The repos cloned under M3PATH has two functions, building images and
+# providing manifest generation functionality
 mkdir -p "${M3PATH}"
+# When building local images make sure FORCE_REPO_UPDATE is set to 'false'
+# otherwise clone_repo will overwrite the content of whatever is at the end
+# of the path
+detect_mismatch "${BMO_LOCAL_IMAGE:-}" "${BMOPATH}"
 clone_repo "${BMOREPO}" "${BMOBRANCH}" "${BMOPATH}" "${BMOCOMMIT}"
+
+detect_mismatch "${CAPM3_LOCAL_IMAGE:-}" "${CAPM3PATH}"
 clone_repo "${CAPM3REPO}" "${CAPM3BRANCH}" "${CAPM3PATH}" "${CAPM3COMMIT}"
+
+detect_mismatch "${IPAM_LOCAL_IMAGE:-}" "${IPAMPATH}"
 clone_repo "${IPAMREPO}" "${IPAMBRANCH}" "${IPAMPATH}" "${IPAMCOMMIT}"
+
+detect_mismatch "${CAPI_LOCAL_IMAGE:-}" "${CAPIPATH}"
 clone_repo "${CAPIREPO}" "${CAPIBRANCH}" "${CAPIPATH}" "${CAPICOMMIT}"
-if [[ "${BUILD_MARIADB_IMAGE_LOCALLY:-}" == "true" ]]; then
+
+# MariaDB and Ironic source is not needed unless the images are built locally
+# If the repo path does not match with the IMAGE location that means the image
+# is built from a repo that is not under dev-env's control thus there is no
+# need to clone the repo.
+# There is no need to keep the PATH and the IMAGE vars in sync as there
+# is no other use of the path variable than cloning
+if [[ "${MARIADB_LOCAL_IMAGE:-}" == "${MARIADB_IMAGE_PATH}" ]]; then
   clone_repo "${MARIADB_IMAGE_REPO}" "${MARIADB_IMAGE_BRANCH}" "${MARIADB_IMAGE_PATH}" "${MARIADB_IMAGE_COMMIT}"
 fi
-if [[ ${IRONIC_FROM_SOURCE:-} == "true" || ${BUILD_IRONIC_IMAGE_LOCALLY:-} == "true" ]]; then
```

Comment:
removing this is quite destructive for upstream ironic CI, especially removing IRONIC_FROM_SOURCE logic
https://opendev.org/openstack/ironic/src/branch/master/playbooks/metal3-ci/run.yaml#L12

Reply:
Same answer as in #1273 IRONIC_FROM_SOURCE logic has not been removed https://github.com/metal3-io/metal3-dev-env/pull/1273#discussion_r1288325054

Reply:
I will add a hold until you confirm that I have not done anything incompatible with upstream ironic CI
/hold

Reply:
please remove the hold if everything seems to be OK